guides/developer/plugins/frontend.md

```diff
@@ -0,0 +1,347 @@
+
+The `scaffold` command creates the most up-to-date schema example, which you
+should use as your template when you develop your plugin. If you are writing
+a plugin that targets an older version of Spinnaker, you may need to refer to
+existing stages for the Spinnaker release to ensure you're following the correct schema.
+
+In general, a Deck stage has these elements:
+
+- A `StageConfig` React component.
+    - This component wraps `FormikStageComponent` and refers to the React `StageForm` component.
+- A `StageForm` React component.
+    - This component provides a set of `FormikFormField` components that
+      encapsulate the configuration options for your stage.
+- A (optional) `validate` function.
+    - The `validate` function validates a stage configuration. When added to a stage object, `validate` is called whenever input changes in the config form.
+- A Stage object that encapsulates the previous components.
+    - This object is passed to the plugin config object and registered
+      when Deck starts.
+
+You can define more than one stage per plugin. When you're ready to register
+your plugin and test in Deck, add all your stages to the `stage` field in your
+plugin `index.ts` file like so:
+
+```javascript
+import { IDeckPlugin } from '@spinnaker/core';
+import { widgetizeStage } from './WidgetizeStage';
+
+export const plugin: IDeckPlugin = {
+  stages: [widgetizeStage],
+};
+```
+
+<!-- FIXME link to a frontend plugin example
+https://github.com/spinnaker-plugin-examples/pf4jStagePlugin  needs to be updated to use the latest architecture
+-->
+
+## Overriding existing components
+
+You can also override existing components within Deck, so long as they have
+an `Overridable` annotation or are registered as an overridable component. In
+this use case, you define your replacement component and then leverage the
+`initialize` method of your plugin object to override the component when it is
+loaded.
+
+For example, if you want to remove the ability to modify Application
+configuration in Deck, you define a component like this:
+
+```javascript
+import React from 'react';
+
+export const InvisibleConfig = () => {
+  return <h1>No config here!</h1>;
+}
+```
+
+In order to override the component, you need to know the Application
+configuration registration key. You can find its definition in the
+GitHub Deck project, [ApplicationConfig.tsx](https://github.com/spinnaker/deck/blob/43a0f56fa85fa1714fef0ba73810c90761f5996d/app/scripts/modules/core/src/application/config/ApplicationConfig.tsx). There, you find the `Overridable` annotation is
+`applicationConfigView`. Then, in your `index.ts` file where you define your plugin object, you override that component in the `initialize` method. For example:
+
+
+```javascript
+import { IDeckPlugin, overrideRegistrationQueue } from '@spinnaker/core';
+import { InvisibleConfig } from './InvisibleConfig';
+
+export const plugin: IDeckPlugin = {
+  initialize: _ => {
```

Comment:
I think `() => { ... }` is more idiomatic.

Reply:
- [ ] stop treating TS like Haskell